astropy/io/fits/hdu/hdulist.py

```diff
@@ -510,6 +510,13 @@ def fileinfo(self, index):
 
         return output
 
+    def copy(self):
+        """
+        Return a shallow copy when calling HDUList.copy().
```

Comment:
I don't think it should be a shallow copy, now that I found a similar discussion for FITS header in #4990.

Reply:
@saimn , ah, I just saw your comment about shallow copy in my emails (they were hidden in the browser). If you think it is okay, then I am officially overruled. :smile: